src/viewer.js

```diff
@@ -143,6 +143,14 @@ export class Viewer {
     /** @const @private {!Object<string, string>} */
     this.params_ = {};
 
+    /** @private {?function(*)} */
+    this.onVisibilityResolve_ = null;
```

Comment:
@dvoytenko a little unclear how we can move this to UserNotificationService, since we added this so that the UserNotificationService can subscribe to this event (when viewer becomes visible). do let me know if it is more appropriate to move this.